pkg/internal/source/kind.go

```diff
@@ -26,70 +28,95 @@ type Kind struct {
 
 	// started may contain an error if one was encountered during startup. If its closed and does not
 	// contain an error, startup and syncing finished.
-	started     chan error
-	startCancel func()
+	mu                 sync.Mutex
+	stopped            bool
```

Comment:
nit: let's please add a bit more godoc for the other fields (the godoc for started seems to be in the wrong place now)